homeassistant/components/google/__init__.py

```diff
@@ -70,17 +72,36 @@
 DATA_INDEX = "google_calendars"
 
 YAML_DEVICES = f"{DOMAIN}_calendars.yaml"
-SCOPES = "https://www.googleapis.com/auth/calendar"
 
 TOKEN_FILE = f".{DOMAIN}.token"
 
+
+class FeatureAccess(Enum):
+    """Class to represent different access scopes."""
+
+    read_only = "https://www.googleapis.com/auth/calendar.readonly"
+    read_write = "https://www.googleapis.com/auth/calendar"
+
+    def __init__(self, scope: str) -> None:
+        """Init instance."""
+        self._scope = scope
+
+    @property
+    def scope(self) -> str:
+        """Google calendar scope for the feature."""
+        return self._scope
+
+
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_CLIENT_ID): cv.string,
                 vol.Required(CONF_CLIENT_SECRET): cv.string,
                 vol.Optional(CONF_TRACK_NEW): cv.boolean,
+                vol.Optional(CONF_CALENDAR_ACCESS, default="read_write"): cv.enum(
```

Comment:
This PR violates ADR-0010 @allenporter l, also if you notice documentation is missing, a PR should not be merged.

How should we proceed? I would vote revert this PR.

Reply:
I was not aware of this. I assume this was done to force old integrations to the new UI flow? If this was to be converted, what would the ACs be?

Reply:
>  I assume this was done to force old integrations to the new UI flow

That is not the sole reason.

> If this was to be converted, what would the ACs be?

The "ACs"?

Reply:
Sorry, the acceptance criteria of the conversion? For example, "The client id/secret and access level should be set via the UI flow"

Reply:
It would need an OAuth flow (config flow). Client ID & secrets are part of the YAML configuration, not the UI.

Just looked at the integration, seeing the settings for the specific calendars, I can see this will become problematic. Let's keep this PR in for now.

Reply:
Thanks for looking closer.  Are we good on documentation? Forgot to update the PR links so sorry about that.

My assumption is this is like https://github.com/home-assistant/core/pull/44529 which I aborted, and I haven't put time into exploring a separate oauth UI. An alternative is we do a hybrid of yaml plus config flow for everything but oauth.

Reply:
Also interestingly this is configuring the oauth scope so it's even more bordelrine.

Reply:
OAuth scopes (variable / conditional) can be handled in an OAuth flow, that is not a problem.

Reply:
Thanks for the reviews guys and sorry that what I thought was a simple PR ended up being a controversial update. I'll try and remember the ADR in the future.

What you guys are doing with Home Assistant is appreciated :)